### DISCUSSION TITLE
How generate the "instruction traces"

### DISCUSSION THREAD

**Discussion by FuLiuwei** [ChampSim/ChampSim#475] [Q&A]
Hi, I have encountered an issue and I need everyone's help: How can I fix a seed and generate an instruction traces using champSim? I have downloaded the spec06 and spec17 traces, but I am unsure how to convert it into an instruction trace. For example, how convert 410.bwaves-s0.txt.xz to instruction traces with a fixed seed.
thank you a lot!

**Comment by ngober** [comment]
What is an "instruction trace"? What is a "seed"? You seem to be referencing a compressed text file, but I don't know what that file is. 

The traces you are likely referencing, which are linked in the README, are the traces that serve as input to the simulator. See [here](https://github.com/ChampSim/ChampSim?tab=readme-ov-file#run-simulation).

**Reply by FuLiuwei** [reply]
thanks a lot firstly.
      I am training my code by using champSim_for_ml [here](https://github.com/Quangmire/ChampSim). However, I am unsure how to convert loadTraces into instruction traces. For example, during training, I use loadTraces 410.bwaves-s0.txt.xz, but during simulation, I need to use instruction traces (maybe it is the list of [pc, address]). What is the process for converting loadTraces into instruction traces?

**Reply by ngober** [reply]
I think you should ask the maintainers of that tool. We do not maintain here the technique or traces described in that repository.
